jwst/tests_nightly/general/conftest.py

```diff
@@ -0,0 +1,53 @@
+import os
+import pytest
+import re
+import requests
+
+
+RE_URL = re.compile('\w+://\S+')
+
+
+def pytest_addoption(parser):
+    parser.addoption('--bigdata', action='store_true',
+                     help='Big local datasets')
+
+
+def check_url(url):
+    """ Determine if `url` can be resolved without error
+    """
+    if RE_URL.match(url) is None:
+        return False
+
+    r = requests.head(url, allow_redirects=True)
+    if r.status_code >= 400:
+        return False
+    return True
+
+
+class BigdataError(Exception):
+    pass
+
+
+@pytest.fixture
+def _bigdata():
+    """ Return path to large data sets
+
+    Note: Support for URLs added for future integrations
+    """
+    origins = [
+        os.environ.get('TEST_BIGDATA', ''),
```

Comment:
Can't we just do `path = os.environ.get('TEST_BIGDATA', '/data4/jwst_test_data')` and just test for it without the following loop?

Reply:
Yeah, it won't hurt to set the default path within `.get`. As for the list and loop -- The list will eventually contain a URL pointing back to Artifactory. The `check_url` function determines whether the remote resource is available or not. Another (unwritten) function in the test framework uses `_bigdata` to copy or download the data to the local directory.

In other words this should allow you to transparently run the test suites on your desktop at work, or directly on the test system. Let's revisit this when I'm done with the second half of it. I must have gotten sidetracked and never finished.